src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -200,10 +204,14 @@ public function has($id)
             if ('service_container' === $id
                 || isset($this->aliases[$id])
                 || isset($this->services[$id])
-                || array_key_exists($id, $this->services)
             ) {
                 return true;
             }
+            if (array_key_exists($id, $this->services)) {
+                @trigger_error(sprintf('Checking for the existence of a unset "%s" service is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0.', $id), E_USER_DEPRECATED);
```

Comment:
hmm no, we were already unsetting it, so we already don't have null values in this array